components/Tabs/Tabs.tsx

```diff
@@ -67,37 +139,101 @@ export const Tabs = ({ children }: TabsProps) => {
     [children]
   );
 
-  const labels = childTabs.map(({ props: { label } }) => label);
+  const dropdownVarsArr = useMemo(() => {
+    const dropdownVars: Set<string> = new Set();
+
+    childTabs.forEach(({ props: { inDropdown } }) => {
+      if (inDropdown) {
+        const dropdownFromItem = getDropdownFromItem(inDropdown);
+        dropdownFromItem.forEach((item) => dropdownVars.add(item));
+      }
+    });
+
+    return Array.from(dropdownVars).sort().concat(FAKE_DROPDOWN);
+  }, [childTabs]);
+
+  // making data which tabs to display in which dropdown options
+  // and which tabs should be selected in every dropdown
+  const tabsInDropdown: TabsInDropdowns = useMemo(() => {
+    const data: TabsInDropdowns = {};
+
+    for (const option of dropdownVarsArr) {
+      data[option] = [];
+      childTabs.forEach(({ props: { label, selected, inDropdown } }) => {
+        let dataTab: DataTab;
+        if (inDropdown && option !== FAKE_DROPDOWN) {
+          if (isInDropdown(inDropdown, option)) {
+            dataTab = { label, isPreSelected: Boolean(selected) };
+          }
+        } else {
+          dataTab = { label, isPreSelected: Boolean(selected) };
+        }
+        if (dataTab) {
+          data[option].push(dataTab);
+        }
+      });
+    }
+
+    return data;
+  }, [childTabs, dropdownVarsArr]);
 
-  const [currentLabel, setCurrentLabel] = useState(getSelectedLabel(childTabs));
+  const [selectedDropdownOption, setSelectedDropdownOpt] = useState(
+    dropdownSelected ? dropdownSelected : dropdownVarsArr[0]
+  );
+  const tabsMeta = tabsInDropdown[selectedDropdownOption];
+  const [currentTab, setCurrentTab] = useState(getSelectedTab(tabsMeta));
+
+  useEffect(() => {
+    setCurrentTab(getSelectedTab(tabsMeta));
+  }, [tabsMeta, selectedDropdownOption]);
```

Comment:
You have an unused dependency `selectedDropdownOption` and two missing dependencies `setCurrentTab` and `getSelectedTab`.